File: xml-java/src/main/java/com/theminesec/example/headless/ClientApp.java
package com.theminesec.example.headless;

import android.app.Application;
import android.util.Log;
import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;
import com.theminesec.sdk.headless.HeadlessSetup;
import com.theminesec.sdk.headless.model.WrappedResult;
import com.theminesec.sdk.headless.model.setup.SdkInitResp;

public class ClientApp extends Application {
    private final MutableLiveData<WrappedResult<SdkInitResp>> _sdkInitStatus = new MutableLiveData<>();
    public LiveData<WrappedResult<SdkInitResp>> sdkInitStatus = _sdkInitStatus;

    @Override
    public void onCreate() {
        super.onCreate();

        HeadlessSetup.INSTANCE.initSoftPosCallBack(
                this,
                "test.license",
                result -> {
                    Log.d(Constants.TAG, "Application init: " + result.toString());
                    _sdkInitStatus.setValue(result);
                    return null;
                }
        );

    }
}

File: app/src/main/java/com/theminesec/example/headless/ClientApp.kt
package com.theminesec.example.headless

import android.app.Application
import android.util.Log
import com.theminesec.example.headless.util.TAG
import com.theminesec.sdk.headless.HeadlessSetup
import com.theminesec.sdk.headless.model.WrappedResult
import com.theminesec.sdk.headless.model.setup.SdkInitResp
import kotlinx.coroutines.CoroutineScope
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.launch

class ClientApp : Application() {
    private val appScope = CoroutineScope(Dispatchers.Main)
    var sdkInitResp: WrappedResult<SdkInitResp>? = null
        private set

    override fun onCreate() {
        super.onCreate()
        appScope.launch {
            val clientAppInitRes = HeadlessSetup.initSoftPos(this@ClientApp, "public-test.license")
            Log.d(TAG, "$clientAppInitRes")
            sdkInitResp = clientAppInitRes
        }
    }
}
